crates/primitives/simulations/Cargo.toml

```diff
@@ -15,8 +15,22 @@ serde = { workspace = true, features = ["derive"], optional = true }
 serde_with = { workspace = true, optional = true }
 
 # Madara primitives
-mp-felt = { workspace = true }
+mp-felt = { workspace = true, features = [
+  "parity-scale-codec",
+  "serde",
+  "scale-info",
+] }
 mp-messages = { workspace = true }
+mp-state = { workspace = true, features = [
+  "parity-scale-codec",
+  "serde",
+  "scale-info",
+] }
+mp-transactions = { workspace = true, features = [
+  "parity-scale-codec",
+  "serde",
+  "scale-info",
+] }
```

Comment:
serde is an optional dependency here.
Same for psc and si.

So you don't have to build with those features by default. They are only build if the corresponding feature is enabled on this crate

Reply:
Add mp-state/<serde/psc/si> in the features bellow. Same for mp-transaction

Reply:
Yes I forgot thanks

Reply:
What I meant is that you should replace this:
```
mp-transactions = { workspace = true, features = [
  "parity-scale-codec",
  "serde",
  "scale-info",
] }
```

by this
```
mp-transactions = { workspace = true }
```
 
 as long as you don
 ```
 parity-scale-codec = [
   ...,
   "mp-transactions/parity-scale-codec",
]
```